src/electionguard-ui/ElectionGuard.UI/ViewModels/CreateTallyViewModel.cs

```diff
@@ -59,13 +60,13 @@ public partial class CreateTallyViewModel : BaseViewModel
     private Manifest _currentManifest;
 
     [ObservableProperty]
-    private List<string> _deviceList = new();
+    private ObservableCollection<string> _deviceList = new();
 
     [ObservableProperty]
-    private List<string> _dateList = new();
+    private ObservableCollection<string> _dateList = new();

```

Comment:
In some cases, a timing issue occurred where the list update is rendered before the list actually has the data. This creates a situation where the index is out of range.